catlin/pkg/cmd/validate/validate.go

```diff
@@ -76,6 +76,11 @@ func validate(cli app.CLI, path string) error {
 		validator.NewContentValidator(cli, res),
 	}
 
+	kindValidator, err := validator.ValidatorForKind(cli, res)
```

Comment:
nit: 
```go
if v, err := validator.ForKind(cli, res.Kind); err == nil {
  validators = append(validators, v)
}
```

On a second thought, it seems like an anti-pattern to ignore error. How about a `NoopValidator` if kind has no specific validator and `err` will a `not nil` if there was a failure to create validator at which point the CLI should exit with that error? 
So .. the code would look like 

```go 
v, err := validator.ForKind(res.Kind)
if err != nil {
  return err
}
validators = append(validators, v)

```

Reply:
When do we want `ForKind` to return an error? we are just checking the kind and returning a validator if exists otherwise we return `NoopValidator`. we aren't doing anything on the resource inside as it is already parsed. so, not sure when we should return an error :thinking:
If res is of  `UnknownKind` the parser would fail and return from there.